src/sdk/pynni/tests/test_compressor_tf.py

```diff
@@ -0,0 +1,103 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+from unittest import TestCase, main
+
+import numpy as np
+import tensorflow as tf
+from tensorflow.keras import Model, Sequential
+from tensorflow.keras.layers import (Conv2D, Dense, Flatten, MaxPool2D)
+
+from nni.compression.tensorflow import LevelPruner
+
+
+####
+#
+# This file tests pruners on 2 models: a classic CNN model, and a naive model with one linear layer
+#
+# The CNN model is used to test layer detecting and instrumenting.
+#
+# The naive model is used to test mask calculation.
+# It has a single 10x10 linear layer without bias, and `reduce_sum` its result.
+# To help predicting pruning result, the linear layer has fixed initial weights:
+#     [ [ 0.0, 1.0, 2.0, ..., 9.0 ], [0.1, 1.1, 2.1, ..., 9.1 ], ... , [0.9, 1.0, 2.9, ..., 9.9 ] ]
+#
+####
+
+
+# This tensor is used as input of 10x10 linear layer, the first dimension is batch size
+tensor1x10 = tf.constant([[1.0] * 10])
+
+pruners = {
+    'level': (lambda model: LevelPruner(model, [{'sparsity': 0.9, 'op_types': ['default']}])),
+}
+
+
+class TfCompressorTestCase(TestCase):
+
+    def test_layer_detection(self):
+        # Conv and dense layers should be compressed, pool and flatten should not.
+        # This also tests instrumenting functionality.
+        self._test_layer_detection_on_model(CnnModel())
+        #self._test_layer_detection_on_model(build_sequential_model())
+
+    def _test_layer_detection_on_model(self, model):
+        pruner = pruners['level'](model)
+        pruner.compress()
+        layer_types = sorted(wrapper.layer_info.type for wrapper in pruner.wrappers)
+        assert layer_types == ['Conv2D', 'Dense', 'Dense'], layer_types
+
+
+    def test_level_pruner(self):
+        # prune 90% : 9.0 + 9.1 + ... + 9.9 = 94.5
+        model = build_naive_model()
+        pruners['level'](model).compress()
+        x = model(tensor1x10)
+        assert x.numpy() == 94.5
+
+
+class CnnModel(Model):
+    def __init__(self):
+        super().__init__()
+        self.conv = Conv2D(filters=10, kernel_size=3, activation='relu')
+        self.pool = MaxPool2D(pool_size=2)
+        self.flatten = Flatten()
+        self.fc1 = Dense(units=10, activation='relu')
+        self.fc2 = Dense(units=1, activation='softmax')
```

Comment:
I suspect the output of `fc2` is always `1` since there must be a `1` if activation is` softmax`.

Reply:
I'll fix it. But since the result of this model is never used, there should be no real difference.
The correctness of mask algorithm is checked by naive model.

Reply:
OK.